src/engine/sync/enginesync.cpp

```diff
@@ -84,144 +149,133 @@ void EngineSync::requestSyncMode(Syncable* pSyncable, SyncMode mode) {
                 syncDeckExists()) {
             // Internal clock cannot be disabled if there are other decks with
             // sync on. Notify them that their mode has not changed.
-            pSyncable->setSyncMode(SYNC_MASTER);
+            pSyncable->setSyncMode(SYNC_MASTER_SOFT);
         } else {
             deactivateSync(pSyncable);
         }
     }
     checkUniquePlayingSyncable();
 }
 
+Syncable* EngineSync::findBpmMatchTarget(Syncable* requester) {
+    Syncable* stoppedTarget = nullptr;
+    bool foundTargetBpm = false;
+
+    for (const auto& pOtherSyncable : m_syncables) {
```

Comment:
qAsConst(m_syncables) to prevent copies

Reply:
this is unnecessary because const auto& iteration does not make a copy.

Reply:
Unfortunately this is not true due to implicit sharing.

@daschuer Do you still have the original links you once sent me?

Reply:
It all depends if the m_syncables is already shared or not. Currently it looks like it is no yet shared, so the code seems to be OK, but it can't hurt to use as const qAsConst() to protect from sharing due to later code changes. 
https://www.dvratil.cz/2015/06/qt-containers-and-c11-range-based-loops/

Reply:
Code should be robust and not have unintended side-effects caused by unrelated future changes. That's the whole point here. The main issue is of course the non-transparent COW magic used by Qt for convenience that is prone to cause such side-effects.

Reply:
I'll include the call. thanks for the pointer!  (Fix coming in a push soon)